pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -439,7 +443,7 @@ public void sendAsync(Message<?> message, SendCallback callback) {
 
             // validate msg-size (For batching this will be check at the batch completion size)
             int compressedSize = compressedPayload.readableBytes();
-            if (compressedSize > ClientCnx.getMaxMessageSize() && !this.conf.isChunkingEnabled()) {
+            if (compressedSize > chunkMaxMessageSize && !this.conf.isChunkingEnabled()) {
```

Comment:
This should not be changed to `chunkMaxMessageSize`, otherwise we won't be able to send message larger than `chunkMaxMessageSize` if chunking is disabled.

Reply:
fixed.